encoding/value/decimal_value.rs

```diff
@@ -352,6 +417,12 @@ mod tests {
 
                 assert_eq!(as_i128(lhs * rhs), bigint_mul_result, "{:?} * {:?} != {:?}", lhs, rhs, lhs * rhs);
             }
+
+            // can only divide by u64
+            let rhs = rng.gen_range(1..=FRACTIONAL_PART_DENOMINATOR);
+            if as_i128(lhs).checked_div(rhs as i128).is_some_and(|res| range.contains(&res)) {
+                assert_eq!(as_i128(lhs / rhs), as_i128(lhs) / rhs as i128, "{:?} / {:?} != {:?}", lhs, rhs, lhs / rhs);
+            }
```

Comment:
The naïve implementation is identical to the one we're checking here, this test only exists to protect against regressions.